src/org/nyanya/android/traditionalt9/DBUpdateService.java

```diff
@@ -49,16 +50,24 @@ protected void onHandleIntent(Intent intent) {
 			return;
 		}
 		Log.d("T9DBUpdate.onHandle", "Update pass check.");
+
 		// do real things
-		Notification notification = new Notification(R.drawable.ime_en_lang_lower, getText(R.string.updating_database_title),
```

Comment:
I couldn't figure out the scenario when the database needs upgrading to trigger this handler. I suppose it works, but I haven't tested it properly. Any help with confirming it would be welcome.

Reply:
I'm not sure what the question is. Are you asking what exact logic path leads to a database upgrade?
https://github.com/Clam-/TraditionalT9/blob/a352deca43c9b81152ddae7d2a551bc1be10b563/src/org/nyanya/android/traditionalt9/DBUpdateService.java#L45
is called from
https://github.com/Clam-/TraditionalT9/blob/a352deca43c9b81152ddae7d2a551bc1be10b563/src/org/nyanya/android/traditionalt9/T9DB.java#L139
is called when this returns true
https://github.com/Clam-/TraditionalT9/blob/a352deca43c9b81152ddae7d2a551bc1be10b563/src/org/nyanya/android/traditionalt9/T9DB.java#L471
which is built upon
https://developer.android.com/reference/android/database/sqlite/SQLiteOpenHelper

If you are asking about the easiest way to test, it would depend on what you want to test. Here are the various upgrades that have happened over the years:
https://github.com/Clam-/TraditionalT9/blob/a352deca43c9b81152ddae7d2a551bc1be10b563/src/org/nyanya/android/traditionalt9/T9DB.java#L526

If you simply want to make that code path run, the easiest way would be to install current version of the app, load a dictionary, then modify ```DATABASE_VERSION = 5```:
https://github.com/Clam-/TraditionalT9/blob/a352deca43c9b81152ddae7d2a551bc1be10b563/src/org/nyanya/android/traditionalt9/T9DB.java#L31
Compile and reinstall and that path should be run the next time the IME is initialized.

Hope this helps.

Reply:
Hey can anyone upload the latest version that works with a delete key instead of back I need to F21 Pro Thanks. And also wanted to know with can get it in file apk Ready to install the latest version of the app Thanks

Reply:
Yes, @Clam-, this helps. I'll try what you suggested.

@wspdyb, this PR is not about the BACK key. I do have a work-in-progress branch, but that's it. It's the first time I am developing for Android, so I am yet to learn how build, sign and publish an .apk.

Reply:
I have  F21 Pro and I need the apk For the keyboard thanks to all the helpers so far